### PR TITLE
[aot] GfxRuntime140 convention docs

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
Issue: #

### Brief Summary

This PR added detailed documentation about the GfxRuntime140 convention.

### CODE REVIEW

File: python/taichi/aot/conventions/gfxruntime140/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# GfxRuntime140 convention
+
+GfxRuntime140 is a legacy AOT module convention that serves the Vulkan, Metal and OpenGL backends.
+
+GfxRuntime140 only accept SPIR-V as the only valid code artifact. For each SPIR-V artifact, all of the following requirements *must* be satisfied.
```
Comment (ailzhang) [inline python/taichi/aot/conventions/gfxruntime140/README.md:5]:
```suggestion
GfxRuntime140 only accepts SPIR-V as the only valid code artifact. For each SPIR-V artifact, all of the following requirements *must* be satisfied.
```

File: python/taichi/aot/conventions/gfxruntime140/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+- SPIR-V module version is 1.3 or higher.
+- There is only one entry point function called `main`. It's execution model is `GLCompute`. Workgroup size Y and Z are always 1.
```
Comment (ailzhang) [inline python/taichi/aot/conventions/gfxruntime140/README.md:8]:
```suggestion
- There is only one entry point function called `main`. It's execution model is `GLCompute`. Workgroup size Y and Z are always `1`.
```

File: python/taichi/aot/conventions/gfxruntime140/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- There is only one descriptor set, where `Set=0`.
+- The context buffer is a uniform bufffer in `std140` layout, bound to `Binding=0`. Its elements are aligned to 4 bytes. The size is at least 1536 bytes. The context buffer content must be consumed following the rules listed below in "Context buffer format".
+- The root buffer is a storage buffer in `std430` layout, bound to `Binding=1`.
+- ND-arrays are storage buffers in `std430` layout, bound to `Binding>=2` in argument order.
+
+## Context buffer layout
+
+The context buffer has the following constituents sequentially in order:
+
+- Argument values. Size of this section depends on the number of arguments and the types of the arguments.
```
Comment (ailzhang) [inline python/taichi/aot/conventions/gfxruntime140/README.md:18]: OOC: is this section trying to describe the runtime context layout or a different one? (IIRC in runtime context all scalars takes 8bytes right?

**Review by ailzhang — COMMENTED** [review]
Thanks!

File: python/taichi/aot/conventions/gfxruntime140/README.md (hunk 3 above)
Comment (PENGUINLIONG) [reply python/taichi/aot/conventions/gfxruntime140/README.md:18]: GfxRuntime seems different, there is a HostDeviceBlitter doing all the copy. I played with it a bit and it seems a 32b argument only take one uint

**Review by ailzhang — APPROVED** [review]

**Pull request merged**